gammapy/irf/core.py

```diff
@@ -38,10 +44,22 @@ class IRF(metaclass=abc.ABCMeta):
         fill_value=0.0,
     )
 
-    def __init__(self, axes, data=0, unit="", meta=None, interp_kwargs=None):
+    def __init__(
+        self,
+        axes,
+        data=0,
+        unit="",
+        is_pointlike=False,
+        fov_alignment=FoVAlignment.RADEC,
+        meta=None,
+        interp_kwargs=None,
+    ):
         axes = MapAxes(axes)
         axes.assert_names(self.required_axes)
         self._axes = axes
+        self.fov_alignment = FoVAlignment(fov_alignment)
```

Comment:
I guess all meta attributes should be read-only as well? I was wondering, whether it's worth to group all meta info in its own namespace? E.g. by introducing an `IRFMeta` class or similar, which is accessible as `IRF.meta`. This keeps the `IRF` base class more general, because I could imaging that the amount of meta information will grow in future, when we introduce e.g. event classes and types...

Reply:
I am not sure about that. Especially at the moment, we cannot really call it `meta`, since that still needs to be the actual FITS header data.

These attributes really belong to the IRF itself, since otherwise the `.data` cannot be interpreted correctly.

Reply:
> I guess all meta attributes should be read-only as well? 

That would mean creating either properties for all of them, or adding a `__setattr__` that raises an error. If we decide to make IRFs immutable (at least semantically, it's not possible to really do that in python), I'd go for the removed `__setattr__` instead of adding all the properties code.

Reply:
See e.g. how dataclasses are implementing the `frozen` parameter:

https://docs.python.org/3/library/dataclasses.html#frozen-instances

Reply:
> I am not sure about that. Especially at the moment, we cannot really call it meta, since that still needs to be the actual FITS header data.

Yes, for now any proposed solution will lead to cleaner code.

> These attributes really belong to the IRF itself, since otherwise the .data cannot be interpreted correctly.

I agree, however the `IRFMeta` can be mandatory as well. As I said for me it's mostly a question of code organization / definition of namespace. I could possibly imagine that meta information could be read and validated independently of the actual data of the IRF...but lm totally fine to postpone this discussion or open a separate issue to discuss it there.

>  See e.g. how dataclasses are implementing the frozen parameter:

Yes, I think technically this is definitely a good solution for an `IRFMeta` class. Applying this pattern to the `IRF` class, I'm not sure yet. I think historically we have worked in Gammapy with properties, but only because the Gammapy code is older than `dataclasses`. However for consistency it might be preferable to continue with properties for now, except if we find us adding new properties in more than 10-15 places.

Reply:
Ok, I added a property for `fov_alignment` and a test ensuring that `setattr` doesn't work for them